code/game/machinery/computer/communications.dm

```diff
@@ -393,6 +395,28 @@ var/shuttle_call/shuttle_calls[0]
 		// auto update every Master Controller tick
 		ui.set_auto_update(1)
 
+/obj/machinery/computer/communications/emag(mob/user as mob)
+	if(!emagged)
+		emagged = 1
+		user << "Syndicate routing data uploaded!"
+		new/obj/effect/effect/sparks(get_turf(src))
+		playsound(loc,"sparks",50,1)
+		authenticated = 2
+		setMenuState(usr,COMM_SCREEN_MAIN)
+		update_icon()
+		return 1
+	return
+
+
+/obj/machinery/computer/communications/update_icon()
+	..()
+	icon_state = "[emagged ? "[initial(icon_state)]-emag" : "[initial(icon_state)]"]"

```

Comment:
If you insist on using initial() please just assign the initial icon_state to a variable either in the proc or on the object, rather than potentially needing to call it twice if it is broken or unpowered.